### PR TITLE
DOP-2174: Add card-tag refrole

### PR DESCRIPTION
**Pull request by mango-db**
[DOP-2174](https://jira.mongodb.org/browse/DOP-2174).

Add card-tag role to represent the Realm landing page's [new Card tag components](https://www.figma.com/file/hPTwMb9uHOypYLckYDg072/Realm-PLP?node-id=0%3A1) (see: `Android ->`, `iOS ->`, etc). We expect the following rST:

```
.. card::
   :options: blah

   Card text.

   :card-tag:`Android <android-intro>`
```

### CODE REVIEW

File: snooty/rstspec.toml [hunk 1]
```diff
@@ -1029,6 +1029,10 @@ type = "text"
 help = """Show a file path."""
 type = "text"
 
+[role.card-ref]
```
Comment (i80and) [inline snooty/rstspec.toml:1032]: Could we:
* Place this in the `mongodb` domain
* Add a help string

What directive will create the targets to be linked to?

**Review by i80and — CHANGES_REQUESTED** [review]

File: snooty/rstspec.toml (hunk 1 above)
Comment (schmalliso) [reply snooty/rstspec.toml:1032]: @i80and Since she's inheriting ref, will this not work to link to refs?

Comment (i80and) [reply snooty/rstspec.toml:1032]: @schmalliso well, it'll link to labels specifically Just making sure that's intended -- this is a new use case to me :D

Comment (schmalliso) [reply snooty/rstspec.toml:1032]: Reasonable! Good, I've not led her completely astray then (that is the desired use-case).

Comment (mango-db) [reply snooty/rstspec.toml:1032]: @i80and I believe there is a help string already, unless you're referring to something else?

Comment (i80and) [reply snooty/rstspec.toml:1032]: @mango-db oh snap you're right -- sorry, I don't know how I missed that :D If we could be slightly more specific about what we mean by "styled reference" and what the intent is, that would be ideal. A help string should be enough for a writer to know when they might want to use it.

**Review by i80and — APPROVED** [review]
🥳

**Pull request merged**